File: code-examples/zig/advanced_integer_overflow.zig
fn add(a: u8, b: u8) u8 {
    return a + b;
}

pub fn main() u8 {
    const a: u8 = 70;
    const b: u8 = 255;
    const x = @as(usize, @intCast(a));
    const y = @as(usize, @intCast(b));
    const z = @as(usize, @intCast(add(a, b)));
    return if (x + y != z) 1 else 0;
}

File: code-examples/rust/advanced_integer_overflow.rs
fn add(a: u8, b: u8) -> u8 {
    a + b
}

pub fn main() -> std::process::ExitCode {
    let a: u8 = 70;
    let b: u8 = 255;
    let x = a as usize;
    let y = b as usize;
    let z = add(a, b) as usize;
    ((x + y != z) as u8).into()
}
